src/vs/workbench/parts/terminal/electron-browser/terminalPanel.ts

```diff
@@ -149,25 +149,35 @@ export class TerminalPanel extends Panel {
 				// occurs on the selection itself.
 				this._terminalService.getActiveInstance().focus();
 			} else if (event.which === 3) {
-				// Trigger the context menu on right click
-				let anchor: HTMLElement | { x: number, y: number } = this._parentDomElement;
-				if (event instanceof MouseEvent) {
-					const standardEvent = new StandardMouseEvent(event);
-					anchor = { x: standardEvent.posx, y: standardEvent.posy };
-				}
+				let rightClickAction: string = this._terminalService.configHelper.getRightClickAction();
+				if (rightClickAction === 'copyPaste') {
+					let terminal = this._terminalService.getActiveInstance();
+					if (terminal.hasSelection()) {
+						terminal.copySelection();
+						terminal.clearSelection();
+					} else {
+						terminal.paste();
+					}
+				} else if (rightClickAction === 'contextMenu') {
+					let anchor: HTMLElement | { x: number, y: number } = this._parentDomElement;
```

Comment:
Would prefer if these were 2 different variables (if you need to store `this._parentDomElement` at all)